app/(main)/account/recovery/page.tsx

```diff
@@ -0,0 +1,261 @@
+// Copyright (c) Gridiron Survivor.
+// Licensed under the MIT License.
+
+'use client';
+import { resetPassword } from '@/api/apiFunctions';
+import Alert from '@/components/AlertNotification/AlertNotification';
+import { AlertVariants } from '@/components/AlertNotification/Alerts.enum';
+import { Button } from '@/components/Button/Button';
+import { Input } from '@/components/Input/Input';
+import LoadingSpinner from '@/components/LoadingSpinner/LoadingSpinner';
+import Logo from '@/components/Logo/Logo';
+import { useAuthContext } from '@/context/AuthContextProvider';
+import logo from '@/public/assets/logo-colored-outline.svg';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useRouter, useSearchParams } from 'next/navigation';
+import React, { useEffect, useState } from 'react';
+import { Control, SubmitHandler, useForm, useWatch } from 'react-hook-form';
+import toast from 'react-hot-toast';
+import { z } from 'zod';
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormMessage,
+} from '../../../../components/Form/Form';
+import { Suspense } from 'react';
+import GlobalSpinner from '@/components/GlobalSpinner/GlobalSpinner';
+
+interface CustomError extends Error {
+  type?: string;
+}
+
+/**
+ * The schema for the login form.
+ * @throws {Error} Throws an error if the email is not a valid email address.
+ * @throws {Error} Throws an error if the email is not provided.
+ * @throws {Error} Throws an error if the email is not a string.
+ */
+const ResetPasswordSchema = z
+  .object({
+    password: z
```

Comment:
Why is password value being set to `z` here?

Reply:
look at the next line .... .string .min .min